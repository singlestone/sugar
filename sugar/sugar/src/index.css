@import "tailwindcss/base";
@import "tailwindcss/components";
@import "tailwindcss/utilities";
@import "./css/_default_variables.css";
@import "./css/_index_old.css";
@import "./css/_buttons.css";

/*
This causes a problem with input placeholder text in the example.
@import "./css/_form_controls.css";
*/

@import "./css/_brand-variables.css";
