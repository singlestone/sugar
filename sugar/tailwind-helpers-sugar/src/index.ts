export * from "./color-helpers";
export * from "./colors-with-opacity";
export * from "./create-sugar-plugin";
export * from "./focus-transition.styles";
export * from "./override-nested-css-property";
export * from "./sugar-color-families.enum";
export * from "./sugar-color-shades.enum";
export * from "./sugar-default-plugin-config";
export * from "./sugar-measurements.enum";
export * from "./sugar-plugin-color-options.interface";
export * from "./sugar-plugin-options.interface";
