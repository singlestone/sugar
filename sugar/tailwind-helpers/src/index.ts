export * from "./color-helpers";
export * from "./colors-with-opacity";
export * from "./create-sugar-plugin";
export * from "./disabled-control-styles";
export * from "./focus-transition.styles";
export * from "./sugar-button-config.interface";
export * from "./sugar-color-families.enum";
export * from "./sugar-color-shades.enum";
export * from "./sugar-default-button-config";
export * from "./sugar-default-focus-config";
export * from "./sugar-default-form-config";
export * from "./sugar-default-measurements-config";
export * from "./sugar-default-plugin-config";
export * from "./sugar-measurements.enum";
export * from "./sugar-plugin-color-options.interface";
export * from "./sugar-plugin-options.interface";
