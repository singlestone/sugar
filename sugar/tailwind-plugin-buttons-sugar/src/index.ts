export * from "./buttons-plugin-options.interface";
export * from "./buttons.plugin";
