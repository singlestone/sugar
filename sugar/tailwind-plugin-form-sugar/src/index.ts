export * from "./forms.plugin";
export * from "./inputs/inputs.plugin";
