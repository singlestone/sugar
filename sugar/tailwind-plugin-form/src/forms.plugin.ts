import {
  type SugarTailwindPlugin,
  createSugarPlugin,
} from "@singlestone/tailwind-helpers-sugar";

import { sugarTextInput } from "./inputs/inputs.plugin";

export const sugarFormsComponents = [sugarTextInput];
export const sugarFormsPlugin: SugarTailwindPlugin =
  createSugarPlugin(sugarFormsComponents);
