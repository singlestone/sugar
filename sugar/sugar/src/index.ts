import {
  createSugarPlugin,
  SugarTailwindPlugin,
} from "@singlestone/tailwind-helpers-sugar";
import { sugarButtonsComponents } from "@singlestone/tailwind-plugin-buttons-sugar";
import { sugarFormsComponents } from "@singlestone/tailwind-plugin-form-sugar";

export const sugarComponents = [
  ...sugarFormsComponents,
  ...sugarButtonsComponents,
];

export const sugarCorePlugin: SugarTailwindPlugin =
  createSugarPlugin(sugarComponents);
