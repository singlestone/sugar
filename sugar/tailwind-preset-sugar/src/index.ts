import { sugarButtonsPlugin } from "@singlestone/tailwind-plugin-buttons-sugar";
import { sugarFormsPlugin } from "@singlestone/tailwind-plugin-form-sugar";
import { TailwindPlugin } from "tailwindcss/plugin";
const options = {
  prefix: "sugar",
};
export { theme } from "@singlestone/tailwind-theme-sugar";
export const plugins: TailwindPlugin[] = [
  sugarButtonsPlugin,
  sugarFormsPlugin,
].map((plugin) => plugin(options));
