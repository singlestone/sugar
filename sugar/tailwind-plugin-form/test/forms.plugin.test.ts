import { SugarPluginOptions } from "@singlestone/tailwind-helpers-sugar";
import { resolve } from "path";
import postcss from "postcss";
import tailwindcss from "tailwindcss";
import { Config } from "tailwindcss/types/config";
import { describe, expect, test } from "vitest";

import { sugarFormsPlugin } from "../src";

const run = (
  config: Partial<Config>,
  pluginOptions: SugarPluginOptions = {}
) => {
  const { currentTestName } = expect.getState();
  config = {
    ...{
      plugins: [sugarFormsPlugin(pluginOptions)],
      corePlugins: { preflight: false },
    },
    ...config,
  };

  return postcss(tailwindcss(config as Config)).process(
    ["@tailwind base;", "@tailwind components;", "@tailwind utilities"].join(
      "\n"
    ),
    {
      from: `${resolve(__filename)}?test=${currentTestName}`,
    }
  );
};

const getClassCreationTest = (
  className: string,
  options: SugarPluginOptions = {}
) => {
  const config: Partial<Config> = {
    safelist: [className],
    theme: {},
  };
  return run(config, options).then((result) => {
    expect(result.css).toContain(`.${className}`);
  });
};

describe("sugarFormsPlugin", () => {
  describe("Plugin options", () => {
    const testPrefix = "test_prefix";
    const options = {
      prefix: testPrefix,
    };
    test("set plugin prefix", () =>
      getClassCreationTest(`${testPrefix}-text-input`, options));
  });

  describe("Static Components", () => {
    const expectedClasses = [["sugar-text-input", "sugar-select-input"]];
    test.each(expectedClasses)("create %s class", (className: string) =>
      getClassCreationTest(className)
    );
  });
});
