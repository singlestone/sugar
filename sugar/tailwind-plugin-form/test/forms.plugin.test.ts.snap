// Vitest Snapshot v1

exports[`sugarFormsPlugin > Plugin options > set plugin prefix 1`] = `Promise {}`;

exports[`sugarFormsPlugin > Static Components > create sugar-text-input class 1`] = `Promise {}`;
